src/session.rs

```diff
@@ -211,9 +219,14 @@
         "Editor session `{}` closed, shutting down associated language servers",
         request.meta.session
     );
-    controllers.retain(|route, controller| {
-        if route.session == request.meta.session {
-            info!("Exit {} in project {}", route.language, route.root);
+    controllers.retain(|routes, controller| {
+        if routes
+            .iter()
```

Comment:
We should add an assertion somewhere that either all routes are from this session, or none is.
Because if only some are, then the behavior is weird.